v2/tailnet_settings.go

```diff
@@ -0,0 +1,73 @@
+package tsclient
+
+import (
+	"context"
+	"net/http"
+)
+
+// TailnetSettingsResource provides an API to view/control settings for a tailnet.
+type TailnetSettingsResource struct {
+	*Client
+}
+
+type (
+	// TailnetSettings represents the current settings of a tailnet.
+	TailnetSettings struct {
+		DevicesApprovalOn      bool `json:"devicesApprovalOn"`
+		DevicesAutoUpdatesOn   bool `json:"devicesAutoUpdatesOn"`
+		DevicesKeyDurationDays int  `json:"devicesKeyDurationDays"` // days before device key expiry
+
+		UsersApprovalOn                        bool                              `json:"usersApprovalOn"`
+		UsersRoleAllowedToJoinExternalTailnets RoleAllowedToJoinExternalTailnets `json:"usersRoleAllowedToJoinExternalTailnets"`
+
+		NetworkFlowLoggingOn        bool `json:"networkFlowLoggingOn"`
+		RegionalRoutingOn           bool `json:"regionalRoutingOn"`
+		PostureIdentityCollectionOn bool `json:"postureIdentityCollectionOn"`
+	}
+
+	// UpdateTailnetSettingsRequest is a request to update the settings of a tailnet.
+	// Nil values indicate that the existing setting should be left unchanged.
+	UpdateTailnetSettingsRequest struct {
+		DevicesApprovalOn      *bool `json:"devicesApprovalOn,omitempty"`
+		DevicesAutoUpdatesOn   *bool `json:"devicesAutoUpdatesOn,omitempty"`
+		DevicesKeyDurationDays *int  `json:"devicesKeyDurationDays,omitempty"` // days before device key expiry
+
+		UsersApprovalOn                        *bool                              `json:"usersApprovalOn,omitempty"`
+		UsersRoleAllowedToJoinExternalTailnets *RoleAllowedToJoinExternalTailnets `json:"usersRoleAllowedToJoinExternalTailnets,omitempty"`
+
+		NetworkFlowLoggingOn        *bool `json:"networkFlowLoggingOn,omitempty"`
+		RegionalRoutingOn           *bool `json:"regionalRoutingOn,omitempty"`
+		PostureIdentityCollectionOn *bool `json:"postureIdentityCollectionOn,omitempty"`
+	}
+
+	// RoleAllowedToJoinExternalTailnets constrains which users are allowed to join external tailnets
+	// based on their role.
+	RoleAllowedToJoinExternalTailnets string
+)
+
+const (
```

Comment:
Nit: add docs on the below three consts

Reply:
I've added links to the API docs and am letting those speak for themselves.